scripts/test-triton.sh

```diff
@@ -181,39 +186,58 @@ run_unit_tests() {
   lit -v . || $TRITON_TEST_IGNORE_ERRORS
 }
 
+run_pytest_command() {
+  local pytest_command=$1
+
+  if [[ -n "$TRITON_TEST_SELECTFILE" ]]; then
+    eval "${pytest_command}" --collect-only > /dev/null 2>&1 && eval "${pytest_command}" || true
+  else
+    eval "${pytest_command}"
+  fi
+}
+
 run_core_tests() {
   echo "***************************************************"
   echo "******      Running Triton Core tests        ******"
   echo "***************************************************"
   cd $TRITON_PROJ/python/test/unit
   ensure_spirv_dis
 
-  TRITON_DISABLE_LINE_INFO=1 TRITON_TEST_SUITE=language \
-    pytest -vvv -n ${PYTEST_MAX_PROCESSES:-8} --device xpu language/ --ignore=language/test_line_info.py --ignore=language/test_subprocess.py --ignore=language/test_warp_specialization.py
+  cmd_language="TRITON_DISABLE_LINE_INFO=1 TRITON_TEST_SUITE=language \
+    pytest -vvv -n ${PYTEST_MAX_PROCESSES:-8} --device xpu language/ --ignore=language/test_line_info.py --ignore=language/test_subprocess.py --ignore=language/test_warp_specialization.py"
+  run_pytest_command "$cmd_language"
```

Comment:
Why so dramatic changes?
```suggestion
  TRITON_DISABLE_LINE_INFO=1 TRITON_TEST_SUITE=language \
    run_pytest_command  -vvv -n ${PYTEST_MAX_PROCESSES:-8} --device xpu language/ --ignore=language/test_line_info.py --ignore=language/test_subprocess.py --ignore=language/test_warp_specialization.py
```

Reply:
Now i get it, you wanted to pass the whole command as a single argument. This will work but ... not very standard)

Reply:
You can make the changes simpler (as i proposed) if you use `"$@"` in run_pytest_command as i proposed below.

Reply:
yep, I got it. In progress